diff --git b/R/add_count.R a/R/add_count.R
index 8fde55c..02da341 100644
--- b/R/add_count.R
+++ a/R/add_count.R
@@ -6,7 +6,7 @@ add_count.duckplyr_df <- function(x, ..., wt = NULL, sort = FALSE, name = NULL,
   x_df <- x
   class(x_df) <- "data.frame"
   out <- add_count(x_df, ..., wt = {{ wt }}, sort = sort, name = name, .drop = .drop)
-  out <- dplyr_reconstruct(out, x)
+  class(out) <- class(x)
   return(out)
 
   # dplyr implementation
