diff --git b/R/anti_join.R a/R/anti_join.R
index b1cad94..e1b81b1 100644
--- b/R/anti_join.R
+++ a/R/anti_join.R
@@ -2,10 +2,20 @@
 #' @importFrom dplyr anti_join
 #' @export
 anti_join.duckplyr_df <- function(x, y, by = NULL, copy = FALSE, ..., na_matches = c("na", "never")) {
+  check_dots_empty0(...)
+  error_call <- caller_env()
+
+  # https://github.com/duckdb/duckdb/issues/6597
+  na_matches <- check_na_matches(na_matches, error_call = error_call)
+
   # Our implementation
   rel_try(
-    "No relational implementation for anti_join()" = TRUE,
+    "Only equi-joins for anti_join()" = inherits(by, "dplyr_join_by") && any(by$condition != "=="),
+    "No relational implementation for anti_join(copy = TRUE)" = copy,
+    # https://github.com/duckdb/duckdb/issues/6597
+    "Can't use anti_join(na_matches = \"na\")" = (na_matches == "na"),
     {
+      out <- rel_filter_join_impl(x, y, by, na_matches, "anti", error_call)
       return(out)
     }
   )
