diff --git b/R/anti_join.R a/R/anti_join.R
index b1cad94..f4df7db 100644
--- b/R/anti_join.R
+++ a/R/anti_join.R
@@ -2,10 +2,17 @@
 #' @importFrom dplyr anti_join
 #' @export
 anti_join.duckplyr_df <- function(x, y, by = NULL, copy = FALSE, ..., na_matches = c("na", "never")) {
+  check_dots_empty0(...)
+  error_call <- caller_env()
+
+  # https://github.com/duckdb/duckdb/issues/6597
+  na_matches <- check_na_matches(na_matches, error_call = error_call)
+
   # Our implementation
   rel_try(
-    "No relational implementation for anti_join()" = TRUE,
+    "No relational implementation for anti_join(copy = TRUE)" = copy,
     {
+      out <- rel_join_impl(x, y, by, "anti", na_matches, error_call = error_call)
       return(out)
     }
   )
