diff --git b/R/arrange.R a/R/arrange.R
index 0c92a1e..d5c6de8 100644
--- b/R/arrange.R
+++ a/R/arrange.R
@@ -2,10 +2,28 @@
 #' @importFrom dplyr arrange
 #' @export
 arrange.duckplyr_df <- function(.data, ..., .by_group = FALSE, .locale = NULL) {
-  # Our implementation
+  force(.data)
+
+  dots <- enquos(...)
+  dots <- unname(dots)
+
+  if (.by_group) {
+    dots <- c(quos(!!!groups(.data)), dots)
+  }
+
   rel_try(
-    "No relational implementation for arrange()" = TRUE,
+    ".locale argument not supported" = !is.null(.locale),
+    "dplyr.legacy_locale not supported" = isTRUE(getOption("dplyr.legacy_locale")),
     {
+      if (length(dots) == 0) {
+        return(.data)
+      }
+      exprs <- rel_translate_dots(dots, .data)
+      rel <- duckdb_rel_from_df(.data)
+      out_rel <- rel_order(rel, exprs)
+      out <- rel_to_df(out_rel)
+      out <- dplyr_reconstruct(out, .data)
+      return(out)
     }
   )
 
