diff --git b/R/arrange.R a/R/arrange.R
index 5688b49..c7530aa 100644
--- b/R/arrange.R
+++ a/R/arrange.R
@@ -2,19 +2,38 @@
 #' @importFrom dplyr arrange
 #' @export
 arrange.duckplyr_df <- function(.data, ..., .by_group = FALSE, .locale = NULL) {
-  # Our implementation
   force(.data)
-  out <- NextMethod()
-  out <- dplyr_reconstruct(out, .data)
-  return(out)
 
-  # dplyr implementation
-  dots <- enquos(...)
+  dots <- unname(enquos(...))
 
   if (.by_group) {
     dots <- c(quos(!!!groups(.data)), dots)
   }
 
+  # Our implementation
+  # Ensure `arrange()` appears in call stack
+  arrange <- rel_try
+  arrange(
+    ".locale argument not supported" = !is.null(.locale),
+    "dplyr.legacy_locale not supported" = isTRUE(getOption("dplyr.legacy_locale")),
+    {
+      if (length(dots) == 0) {
+        return(.data)
+      }
+      exprs <- rel_translate_dots(dots, .data)
+      rel <- duckdb_rel_from_df(.data)
+      out_rel <- rel_order(rel, exprs)
+      out <- rel_to_df(out_rel)
+    },
+    fallback = {
+      out <- NextMethod()
+    }
+  )
+
+  out <- dplyr_reconstruct(out, .data)
+  return(out)
+
+  # dplyr implementation
   loc <- arrange_rows(.data, dots = dots, locale = .locale)
   dplyr_row_slice(.data, loc)
 }
