diff --git b/R/collect.R a/R/collect.R
index aabc60b..eac4dec 100644
--- b/R/collect.R
+++ a/R/collect.R
@@ -2,23 +2,8 @@
 #' @importFrom dplyr collect
 #' @export
 collect.duckplyr_df <- function(x, ...) {
-  # Our implementation
-  rel_try(
-    "No relational implementation for collect()" = TRUE,
-    {
-      return(out)
-    }
-  )
-
-  # dplyr forward
-  x_df <- x
-  class(x_df) <- setdiff(class(x_df), "duckplyr_df")
-  collect <- dplyr:::collect.data.frame
-  out <- collect(x_df, ...)
-  out <- dplyr_reconstruct(out, x)
-  return(out)
-
-  # dplyr implementation
+  # Side effect: ALTREP materialization is triggered
+  nrow(x)
   x
 }
 
