diff --git b/R/count.R a/R/count.R
index 4d53360..b1c1946 100644
--- b/R/count.R
+++ a/R/count.R
@@ -2,6 +2,30 @@
 #' @importFrom dplyr count
 #' @export
 count.duckplyr_df <- function(x, ..., wt = NULL, sort = FALSE, name = NULL, .drop = group_by_drop_default(x)) {
+  force(x)
+
+  dplyr_local_error_call()
+
+  quos <- enquos(...)
+  exprs <- unname(map(quos, quo_get_expr))
+  is_name <- map_lgl(exprs, is_symbol)
+
+  if (all(is_name) && .drop && !sort) {
+    by_chr <- map_chr(exprs, as_string)
+    name <- check_name(name, by_chr)
+
+    if (!(name %in% by_chr)) {
+      n <- tally_n(x, {{ wt }})
+
+      out <- summarise(x, !!name := !!n, .by = c(!!!exprs))
+      out <- dplyr_reconstruct(out, x)
+      return(out)
+    }
+  }
+
+  # FIXME: optimize, no need to forward dots
+  # out <- count(x_df, !!!quos, wt = {{ wt }}, sort = sort, name = name, .drop = .drop)
+
   # Our implementation
   x_df <- x
   class(x_df) <- "data.frame"
@@ -10,8 +34,6 @@ count.duckplyr_df <- function(x, ..., wt = NULL, sort = FALSE, name = NULL, .dro
   return(out)
 
   # dplyr implementation
-  dplyr_local_error_call()
-
   if (!missing(...)) {
     out <- group_by(x, ..., .add = TRUE, .drop = .drop)
   } else {
