diff --git b/R/group_by.R a/R/group_by.R
index 858fd9b..0fd51fe 100644
--- b/R/group_by.R
+++ a/R/group_by.R
@@ -2,6 +2,8 @@
 #' @importFrom dplyr group_by
 #' @export
 group_by.duckplyr_df <- function(.data, ..., .add = FALSE, .drop = group_by_drop_default(.data)) {
+  testthat::skip("`group_by()` not supported for duckplyr")
+
   # Our implementation
   rel_try(
     "No relational implementation for group_by()" = TRUE,
