diff --git b/R/mutate.R a/R/mutate.R
index e995a26..b9f0de3 100644
--- b/R/mutate.R
+++ a/R/mutate.R
@@ -2,10 +2,22 @@
 #' @importFrom dplyr mutate
 #' @export
 mutate.duckplyr_df <- function(.data, ..., .by = NULL, .keep = c("all", "used", "unused", "none"), .before = NULL, .after = NULL) {
+  by <- enquo(.by)
+
   # Our implementation
   rel_try(
-    "No relational implementation for mutate()" = TRUE,
+    "No relational implementation for windowed mutate()" = !quo_is_null(by),
     {
+      rel <- duckdb_rel_from_df(.data)
+      dots <- dplyr_quosures(...)
+
+      keeps <- imap(set_names(names(.data)), relexpr_reference, rel = NULL)
+      mutations <- rel_translate_dots(dots, .data)
+
+      out_rel <- rel_project(rel, c(keeps, mutations))
+      out <- rel_to_df(out_rel)
+      class(out) <- class(.data)
+
       return(out)
     }
   )
