diff --git b/R/nest_join.R a/R/nest_join.R
index ea38000..be1da68 100644
--- b/R/nest_join.R
+++ a/R/nest_join.R
@@ -2,6 +2,16 @@
 #' @importFrom dplyr nest_join
 #' @export
 nest_join.duckplyr_df <- function(x, y, by = NULL, copy = FALSE, keep = NULL, name = NULL, ..., na_matches = c("na", "never"), unmatched = "drop") {
+  # from dplyr implementation
+  check_keep(keep)
+  na_matches <- check_na_matches(na_matches)
+
+  if (is.null(name)) {
+    name <- as_label(enexpr(y))
+  } else {
+    check_string(name)
+  }
+
   # Our implementation
   rel_try(
     "No relational implementation for nest_join()" = TRUE,
