diff --git b/R/pull.R a/R/pull.R
index 2df1e14..350a166 100644
--- b/R/pull.R
+++ a/R/pull.R
@@ -2,10 +2,25 @@
 #' @importFrom dplyr pull
 #' @export
 pull.duckplyr_df <- function(.data, var = -1, name = NULL, ...) {
-  # Our implementation
+  # dplyr implementation
+  var <- tidyselect::vars_pull(names(.data), !!enquo(var))
+  name <- enquo(name)
+  if (!quo_is_null(name)) {
+    name <- tidyselect::vars_pull(names(.data), !!name)
+    var <- c(name, var)
+  }
+
+  loc <- set_names(match(var, names(.data)), var)
+
+  exprs <- exprs_from_loc(.data, loc)
+
   rel_try(
-    "No relational implementation for pull()" = TRUE,
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
     {
+      rel <- duckdb_rel_from_df(.data)
+      out_rel <- rel_project(rel, exprs)
+      out <- rel_to_df(out_rel)
+      out <- deframe(out)
       return(out)
     }
   )
