diff --git b/R/relocate.R a/R/relocate.R
index caaa263..eb40798 100644
--- b/R/relocate.R
+++ a/R/relocate.R
@@ -2,6 +2,28 @@
 #' @importFrom dplyr relocate
 #' @export
 relocate.duckplyr_df <- function(.data, ..., .before = NULL, .after = NULL) {
+  loc <- eval_relocate(
+    expr = expr(c(...)),
+    data = .data,
+    before = enquo(.before),
+    after = enquo(.after),
+    before_arg = ".before",
+    after_arg = ".after"
+  )
+
+  exprs <- exprs_from_loc(.data, loc)
+
+  # Ensure `relocate()` appears in call stack
+  rel_try(
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
+    {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
+      return(out)
+    }, fallback = {
+    }
+  )
+
   # Our implementation
   x_df <- .data
   class(x_df) <- setdiff(class(x_df), "duckplyr_df")
