diff --git b/R/rename.R a/R/rename.R
index 29c398e..971ff4f 100644
--- b/R/rename.R
+++ a/R/rename.R
@@ -2,10 +2,19 @@
 #' @importFrom dplyr rename
 #' @export
 rename.duckplyr_df <- function(.data, ...) {
-  # Our implementation
+  loc <- tidyselect::eval_rename(expr(c(...)), .data)
+  # eval_rename() only returns changes
+  names <- names(.data)
+  names[loc] <- names(loc)
+
+  dupes <- duplicated(loc, fromLast = TRUE)
+  exprs <- exprs_from_loc(.data, loc[!dupes])
+
   rel_try(
-    "No relational implementation for rename()" = TRUE,
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
     {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
       return(out)
     }
   )
