diff --git b/R/rename.R a/R/rename.R
index 910b3f4..4c5f911 100644
--- b/R/rename.R
+++ a/R/rename.R
@@ -2,7 +2,24 @@
 #' @importFrom dplyr rename
 #' @export
 rename.duckplyr_df <- function(.data, ...) {
+  loc <- tidyselect::eval_rename(expr(c(...)), .data)
+  # eval_rename() only returns changes
+  names <- names(.data)
+  names[loc] <- names(loc)
+
   # Our implementation
+  exprs <- exprs_from_loc(.data, names)
+
+  rel_try(
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
+    {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
+      return(out)
+    }, fallback = {
+    }
+  )
+
   x_df <- .data
   class(x_df) <- "data.frame"
   out <- rename(x_df, ...)
@@ -10,7 +27,6 @@ rename.duckplyr_df <- function(.data, ...) {
   return(out)
 
   # dplyr implementation
-  loc <- tidyselect::eval_rename(expr(c(...)), .data)
   # eval_rename() only returns changes
   names <- names(.data)
   names[loc] <- names(loc)
