diff --git b/R/rename.R a/R/rename.R
index 59e5a80..14e1739 100644
--- b/R/rename.R
+++ a/R/rename.R
@@ -2,10 +2,21 @@
 #' @importFrom dplyr rename
 #' @export
 rename.duckplyr_df <- function(.data, ...) {
-  # Our implementation
+  loc <- tidyselect::eval_rename(expr(c(...)), .data)
+  dupes <- duplicated(loc, fromLast = TRUE)
+  loc <- loc[!dupes]
+
+  # eval_rename() only returns changes
+  proj <- rlang::set_names(seq_along(.data), names(.data))
+  names(proj)[loc] <- names(loc)
+
+  exprs <- exprs_from_loc(.data, proj)
+
   rel_try(
-    "No relational implementation for rename()" = TRUE,
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
     {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
       return(out)
     }
   )
