diff --git b/R/rows_update.R a/R/rows_update.R
index 21886430..38c1dad5 100644
--- b/R/rows_update.R
+++ a/R/rows_update.R
@@ -2,7 +2,7 @@
 #' @export
 rows_update.duckplyr_df <- function(x, y, by = NULL, ..., unmatched = c("error", "ignore"), copy = FALSE, in_place = FALSE) {
   # Our implementation
-  rel_try(
+  rel_try(NULL,
     "No relational implementation for rows_update()" = TRUE,
     {
       return(out)
