diff --git b/R/rows_delete.R a/R/rows_delete.R
index 54a42196..051eaa56 100644
--- b/R/rows_delete.R
+++ a/R/rows_delete.R
@@ -2,7 +2,7 @@
 #' @export
 rows_delete.duckplyr_df <- function(x, y, by = NULL, ..., unmatched = c("error", "ignore"), copy = FALSE, in_place = FALSE) {
   # Our implementation
-  rel_try(
+  rel_try(NULL,
     "No relational implementation for rows_delete()" = TRUE,
     {
       return(out)
