diff --git b/R/sample_n.R a/R/sample_n.R
index 759f70f..da3c334 100644
--- b/R/sample_n.R
+++ a/R/sample_n.R
@@ -16,7 +16,7 @@ sample_n.duckplyr_df <- function(tbl, size, replace = FALSE, weight = NULL, .env
 
   # dplyr implementation
   if (!is_null(.env)) {
-    inform("`sample_n()` argument `.env` is deprecated and no longer has any effect.")
+    inform("`sample_n() argument `.env` is deprecated and no longer has any effect.")
   }
 
   size <- enquo(size)
@@ -27,7 +27,6 @@ sample_n.duckplyr_df <- function(tbl, size, replace = FALSE, weight = NULL, .env
     size <- check_size(!!size, n(), replace = replace)
     sample.int(n(), size, replace = replace, prob = !!weight)
   }))
-
 }
 
 duckplyr_sample_n <- function(tbl, ...) {
