diff --git b/R/select.R a/R/select.R
index 2b248c8..f42e978 100644
--- b/R/select.R
+++ a/R/select.R
@@ -2,9 +2,35 @@
 #' @importFrom dplyr select
 #' @export
 select.duckplyr_df <- function(.data, ...) {
-  # Our implementation
   force(.data)
-  out <- NextMethod()
+
+  error_call <- dplyr_error_call()
+
+  loc <- tidyselect::eval_select(
+    expr(c(...)),
+    data = .data,
+    error_call = error_call
+  )
+
+  # Our implementation
+  loc_name <- names(.data)[loc]
+  exprs <- map2(names(.data)[loc], names(loc), ~ expr_reference(.x, alias = .y))
+
+  # Ensure `select()` appears in call stack
+  select <- rel_try
+  select(
+    "Can't use relational with zero-column result set." = (length(loc) == 0),
+    {
+      rel <- duckdb_rel_from_df(.data)
+    }, fallback = {
+      out <- NextMethod()
+      out <- dplyr_reconstruct(out, .data)
+      return(out)
+    }
+  )
+
+  out_rel <- rel_project(rel, exprs)
+  out <- rel_to_df(out_rel)
   out <- dplyr_reconstruct(out, .data)
   return(out)
 
