diff --git b/R/select.R a/R/select.R
index c01975b..bf718b3 100644
--- b/R/select.R
+++ a/R/select.R
@@ -2,6 +2,29 @@
 #' @importFrom dplyr select
 #' @export
 select.duckplyr_df <- function(.data, ...) {
+  force(.data)
+
+  error_call <- dplyr_error_call()
+
+  loc <- tidyselect::eval_select(
+    expr(c(...)),
+    data = .data,
+    error_call = error_call
+  )
+
+  exprs <- exprs_from_loc(.data, loc)
+
+  rel_try(
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
+    {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
+      return(out)
+    }, fallback = {
+      # FIXME: Remove fallback argument
+    }
+  )
+
   # Our implementation
   x_df <- .data
   class(x_df) <- setdiff(class(x_df), "duckplyr_df")
@@ -18,6 +41,7 @@ select.duckplyr_df <- function(.data, ...) {
     data = .data,
     error_call = error_call
   )
+  loc <- ensure_group_vars(loc, .data, notify = TRUE)
 
   # duckplyr: Backend-specific
   loc <- ensure_group_vars(loc, .data, notify = TRUE)
