diff --git b/R/select.R a/R/select.R
index 7015081..45bba33 100644
--- b/R/select.R
+++ a/R/select.R
@@ -2,10 +2,24 @@
 #' @importFrom dplyr select
 #' @export
 select.duckplyr_df <- function(.data, ...) {
-  # Our implementation
+  force(.data)
+
+  error_call <- dplyr_error_call()
+
+  loc <- tidyselect::eval_select(
+    expr(c(...)),
+    data = .data,
+    error_call = error_call
+  )
+
+  exprs <- exprs_from_loc(.data, loc)
+
   rel_try(
-    "No relational implementation for select()" = TRUE,
+    "Can't use relational with zero-column result set." = (length(exprs) == 0),
     {
+      rel <- duckdb_rel_from_df(.data)
+      out <- exprs_project(rel, exprs, .data)
+      return(out)
     }
   )
 
@@ -25,6 +39,7 @@ select.duckplyr_df <- function(.data, ...) {
     data = .data,
     error_call = error_call
   )
+  loc <- ensure_group_vars(loc, .data, notify = TRUE)
 
   # duckplyr: Backend-specific
   loc <- ensure_group_vars(loc, .data, notify = TRUE)
