diff --git b/R/summarise.R a/R/summarise.R
index c15f56c..7c7bac4 100644
--- b/R/summarise.R
+++ a/R/summarise.R
@@ -2,10 +2,20 @@
 #' @importFrom dplyr summarise
 #' @export
 summarise.duckplyr_df <- function(.data, ..., .by = NULL, .groups = NULL) {
-  # Our implementation
+  force(.data)
+
   rel_try(
-    "No relational implementation for summarise()" = TRUE,
     {
+      rel <- duckdb_rel_from_df(.data)
+      dots <- dplyr_quosures(...)
+      aggregates <- rel_translate_dots(dots, .data)
+      by <- eval_select_by(enquo(.by), .data)
+      groups <- lapply(by, relexpr_reference)
+
+      out_rel <- rel_aggregate(rel, groups, aggregates)
+      out <- rel_to_df(out_rel)
+      class(out) <- class(.data)
+
       return(out)
     }
   )
