diff --git b/R/summarise.R a/R/summarise.R
index 88128cd..3839868 100644
--- b/R/summarise.R
+++ a/R/summarise.R
@@ -2,9 +2,32 @@
 #' @importFrom dplyr summarise
 #' @export
 summarise.duckplyr_df <- function(.data, ..., .by = NULL, .groups = NULL) {
+  force(.data)
+
+  rel_try(
+    {
+      rel <- duckdb_rel_from_df(.data)
+      dots <- dplyr_quosures(...)
+      aggregates <- rel_translate_dots(dots, .data)
+      by <- eval_select_by(enquo(.by), .data)
+      groups <- lapply(by, relexpr_reference)
+
+      out_rel <- rel_aggregate(rel, groups, aggregates)
+      out <- rel_to_df(out_rel)
+      class(out) <- class(.data)
+
+      # Side effect: fail if vectors are named
+      out <- as_duckplyr_df(out)
+
+      return(out)
+    },
+    fallback = {
+    }
+  )
+
   # Our implementation
   x_df <- .data
-  class(x_df) <- "data.frame"
+  class(x_df) <- setdiff(class(x_df), "duckplyr_df")
   out <- summarise(x_df, ..., .by = {{ .by }}, .groups = .groups)
   class(out) <- class(.data)
   return(out)
