diff --git b/R/summarise.R a/R/summarise.R
index fb60bdd..684cc39 100644
--- b/R/summarise.R
+++ a/R/summarise.R
@@ -2,10 +2,25 @@
 #' @importFrom dplyr summarise
 #' @export
 summarise.duckplyr_df <- function(.data, ..., .by = NULL, .groups = NULL) {
-  # Our implementation
+  force(.data)
+
   rel_try(
-    "No relational implementation for summarise()" = TRUE,
     {
+      rel <- duckdb_rel_from_df(.data)
+      dots <- dplyr_quosures(...)
+      dots <- fix_auto_name(dots)
+      aggregates <- rel_translate_dots(dots, .data)
+      by <- eval_select_by(enquo(.by), .data)
+      groups <- lapply(by, relexpr_reference)
+
+      out_rel <- rel_aggregate(rel, groups, unname(aggregates))
+      # https://github.com/duckdb/duckdb/issues/7095
+      if (length(groups) == 0) {
+        out_rel <- rel_distinct(out_rel)
+      }
+      out <- rel_to_df(out_rel)
+      class(out) <- class(.data)
+
       return(out)
     }
   )
