diff --git b/R/tbl_vars.R a/R/tbl_vars.R
index 52e7ebd..a046a45 100644
--- b/R/tbl_vars.R
+++ a/R/tbl_vars.R
@@ -4,19 +4,12 @@
 tbl_vars.duckplyr_df <- function(x) {
   # Our implementation
   rel_try(
-    "No relational implementation for tbl_vars()" = TRUE,
     {
-      return(out)
+      rel <- duckdb:::rel_from_altrep_df(x)
+      return(rel_names(rel))
     }
   )
 
-  # dplyr forward
-  x_df <- x
-  class(x_df) <- setdiff(class(x_df), "duckplyr_df")
-  tbl_vars <- dplyr:::tbl_vars.data.frame
-  out <- tbl_vars(x_df)
-  return(out)
-
   # dplyr implementation
   names(x)
 }
