diff --git b/R/transmute.R a/R/transmute.R
index 63a60ce..fab84ac 100644
--- b/R/transmute.R
+++ a/R/transmute.R
@@ -2,7 +2,26 @@
 #' @importFrom dplyr transmute
 #' @export
 transmute.duckplyr_df <- function(.data, ...) {
+  force(.data)
+
+  dots <- check_transmute_args(...)
+  dots <- dplyr_quosures(!!!dots)
+
   # Our implementation
+  rel_try(
+    "Can't use relational with zero-column result set." = (length(dots) == 0),
+    {
+      exprs <- rel_translate_dots(dots, .data)
+      rel <- duckdb_rel_from_df(.data)
+      out_rel <- rel_project(rel, exprs)
+      out <- rel_to_df(out_rel)
+      out <- dplyr_reconstruct(out, .data)
+      return(out)
+    },
+    fallback = {
+    }
+  )
+
   x_df <- .data
   class(x_df) <- "data.frame"
   out <- transmute(x_df, ...)
