diff --git b/R/union.R a/R/union.R
--- b/R/union.R
+++ a/R/union.R
@@ -2,12 +2,10 @@
 #' @export
 union.duckplyr_df <- function(x, y, ...) {
   # Our implementation
-  duckplyr_error <- rel_try(NULL,
-    "No relational implementation for {.code union()}" = TRUE,
-    {
-      return(out)
-    }
-  )
+  check_dots_empty()
+
+  # This is difficult to do manually due to order preservation
+  return(distinct(union_all(x, y)))
 
   # dplyr forward
   check_lazy(x, duckplyr_error)
