diff --git b/R/union.R a/R/union.R
index fda1161..4ecad12 100644
--- b/R/union.R
+++ a/R/union.R
@@ -3,12 +3,10 @@
 #' @export
 union.duckplyr_df <- function(x, y, ...) {
   # Our implementation
-  rel_try(
-    "No relational implementation for union()" = TRUE,
-    {
-      return(out)
-    }
-  )
+  check_dots_empty()
+
+  # This is difficult to do manually due to order preservation
+  return(distinct(union_all(x, y)))
 
   # dplyr forward
   union <- dplyr:::union.data.frame
